parquet-column/src/test/java/org/apache/parquet/column/values/bloom/Bloom.java

```diff
@@ -0,0 +1,413 @@
+    return bitset.length;
+  }
+
+  /**
+   * Compute hash for int value by using its plain encoding result.
+   * @param value the value to hash
+   * @return hash result
+   */
+  public long hash(int value) {
+    ByteBuffer plain = ByteBuffer.allocate(Integer.SIZE/Byte.SIZE);
+    plain.order(ByteOrder.LITTLE_ENDIAN).putInt(value);
+    switch (hash) {
+      case MURMUR3_X64_128: return Murmur3.hash64(plain.array());
+      default:
+        throw new RuntimeException("Not support hash strategy");
+    }
+  }
+
+  /**
+   * Compute hash for long value by using its plain encoding result.
+   * @param value the value to hash
+   * @return hash result
+   */
+  public long hash(long value) {
+    ByteBuffer plain = ByteBuffer.allocate(Long.SIZE/Byte.SIZE);
+    plain.order(ByteOrder.LITTLE_ENDIAN).putLong(value);
+    switch (hash) {
+      case MURMUR3_X64_128: return Murmur3.hash64(plain.array());
+      default:
+        throw new RuntimeException("Not support hash strategy");
+    }
+  }
+
+  /**
+   * Compute hash for double value by using its plain encoding result.
+   * @param value the value to hash
+   * @return hash result
+   */
+  public long hash(double value) {
+    ByteBuffer plain = ByteBuffer.allocate(Double.SIZE/Byte.SIZE);
+    plain.order(ByteOrder.LITTLE_ENDIAN).putDouble(value);
+    switch (hash) {
+      case MURMUR3_X64_128: return Murmur3.hash64(plain.array());
+      default:
+        throw new RuntimeException("Not support hash strategy");
+    }
+  }
+
+  /**
+   * Compute hash for float value by using its plain encoding result.
+   * @param value the value to hash
+   * @return hash result
+   */
+  public long hash(float value) {
+    ByteBuffer plain = ByteBuffer.allocate(Float.SIZE/Byte.SIZE);
+    plain.order(ByteOrder.LITTLE_ENDIAN).putFloat(value);
+    switch (hash) {
+      case MURMUR3_X64_128: return Murmur3.hash64(plain.array());
+      default:
+        throw new RuntimeException("Not support hash strategy");
+    }
+  }
+
+  /**
+   * Compute hash for Binary value by using its plain encoding result.
+   * @param value the value to hash
+   * @return hash result
+   */
+  public long hash(Binary value) {
+    try {
+      ByteBuffer plain = ByteBuffer.allocate(Integer.SIZE/Byte.SIZE);
+      plain.order(ByteOrder.LITTLE_ENDIAN).putInt(value.length());
+      ByteArrayOutputStream baos = new ByteArrayOutputStream(value.length() + 4);
+      baos.write(plain.array(), 0, 4);
+      value.writeTo(baos);
+      switch (hash) {
+        case MURMUR3_X64_128:
+          return Murmur3.hash64(baos.toByteArray());
+        default:
+          throw new RuntimeException("Not support hash strategy");
+      }
+    } catch (IOException e) {
+      throw new ParquetEncodingException("could not insert Binary value to bloom ", e);
+    }
+  }
+
+  /**
+   * Insert element to set represented by bloom bitset.
+   * @param value the value to insert into bloom filter..
+   */
+  public void insert(long value) {
```

Comment:
This requires the hash of a value, not a value.

Reply:
Done.